platform/darwin/src/NSExpression+MGLAdditions.mm

```diff
@@ -665,6 +692,26 @@ - (id)mgl_jsonExpressionObject {
                     [expressionObject addObject:obj.mgl_jsonExpressionObject];
                 }];
                 [expressionObject addObject:self.operand.mgl_jsonExpressionObject];
+                return expressionObject;
+            } else if ([function isEqualToString:@"mgl_matchWithOptions:default:"]) {
+                NSMutableArray *expressionObject = [NSMutableArray arrayWithObjects:@"match", self.operand.mgl_jsonExpressionObject, nil];
+                NSArray *optionsArray = self.arguments[0].constantValue;
+                for (NSDictionary<id, NSExpression*> *options in optionsArray) {
+                    for (NSNumber *key in options.allKeys) {
+                        [expressionObject addObject:key];
+                        [expressionObject addObject:[options[key] mgl_jsonExpressionObject]];
+                    }
+                }
+                
+                [expressionObject addObject:[self.arguments[1] mgl_jsonExpressionObject]];
+                return expressionObject;
+            } else if ([function isEqualToString:@"mgl_coalesce"]) {
```

Comment:
This function needs a stub implementation in Objective-C, even if it’s just an assertion like the one for `-[NSPredicate(MGLAdditions) mgl_matchWithOptions:default:]`. Even if we don’t fully implement these functions in Objective-C, the developer may expect to be able to call `-expressionValueWithObject:context:`. As long as the function name allows for the right number of arguments, we can then raise an exception instead of allowing NSExpression’s parser to segfault.